pages/place/routers/confirm.js

```diff
@@ -27,46 +27,49 @@ module.exports = settings => {
 
   app.post('/', updateDataFromTask(sendData));
 
-  app.use(form(Object.assign({
-    model: 'place',
-    schema: declarationsSchema,
-    saveValues: (req, res, next) => {
-      delete req.session.form[req.model.id].values.declarations;
-      next();
-    },
-    locals: (req, res, next) => {
-      Object.assign(res.locals, { model: req.model });
-      Promise.all([
-        getEstablishment(req),
-        getNacwoById(req, req.form.values.nacwo)
-      ])
-        .then(([establishment, nacwo]) => {
-          Object.assign(res.locals.static, {
-            establishment,
-            schema: Object.assign({}, schema, declarationsSchema),
-            values: {
-              ...req.session.form[req.model.id].values,
-              nacwo
-            }
-          });
-        })
-        .then(() => next())
-        .catch(next);
-    },
-    checkSession: (req, res, next) => {
-      if (req.session.form && req.session.form[req.model.id]) {
-        return next();
+  app.use((req, res, next) => {
+    const requiresDeclaration = !req.user.profile.asruUser;
+    form(Object.assign({
+      model: 'place',
+      schema: requiresDeclaration ? declarationsSchema : {},
+      saveValues: (req, res, next) => {
+        delete req.session.form[req.model.id].values.declarations;
+        next();
+      },
+      locals: (req, res, next) => {
+        Object.assign(res.locals, { model: req.model });
+        Promise.all([
+          getEstablishment(req),
+          getNacwoById(req, req.form.values.nacwo)
+        ])
+          .then(([establishment, nacwo]) => {
+            Object.assign(res.locals.static, {
+              establishment,
+              schema: Object.assign({}, schema, requiresDeclaration ? declarationsSchema : {}),
+              values: {
+                ...req.session.form[req.model.id].values,
+                nacwo
+              }
+            });
+          })
+          .then(() => next())
+          .catch(next);
+      },
+      checkSession: (req, res, next) => {
+        if (req.session.form && req.session.form[req.model.id]) {
+          return next();
+        }
+        return res.redirect(req.buildRoute(settings.page));
+      },
+      editAnswers: (req, res, next) => {
+        delete req.session.form[req.model.id].validationErrors;
+        return res.redirect(req.buildRoute(settings.page));
+      },
+      cancelEdit: (req, res, next) => {
+        return res.redirect(req.buildRoute('place.list'));
       }
-      return res.redirect(req.buildRoute(settings.page));
-    },
-    editAnswers: (req, res, next) => {
-      delete req.session.form[req.model.id].validationErrors;
-      return res.redirect(req.buildRoute(settings.page));
-    },
-    cancelEdit: (req, res, next) => {
-      return res.redirect(req.buildRoute('place.list'));
-    }
-  }, settings)));
+    }, settings))(req, res, next);
```

Comment:
I really hate the pattern of middlewares that call other middlewares.

Could we set `req.requiresDeclaration` in a previous step, and get rid of that?

Reply:
Alternatively make `requiresDeclaration` a function that takes the request as an argument and returns a `Boolean`

Reply:
don't be a hater

Reply:
also yes, although the other 2 corresponding PRs do it this way, so I guess you want them updating too?